File: src/chipschallenge/destroybehaviors/ToggleWallDestroyBehavior.java
package chipschallenge.destroybehaviors;

import chipschallenge.Block;
import chipschallenge.Buttons;

public class ToggleWallDestroyBehavior implements DestroyBehavior {

    private ToggleWallDestroyBehavior() {
    }
    private static ToggleWallDestroyBehavior mInstance = null;

    public static synchronized ToggleWallDestroyBehavior getInstance() {
        if (mInstance == null) {
            mInstance = new ToggleWallDestroyBehavior();
        }
        return mInstance;
    }

    public void destroy(Block b) {
        Buttons.removeGreenButtonsListener(b);
    }
}

File: ChipsChallenge.Shared/Destroybehaviors/ToggleWallDestroyBehavior.cs
﻿namespace ChipsChallenge.Shared.Destroybehaviors
{
    public class ToggleWallDestroyBehavior : IDestroyBehavior
    {
        private static ToggleWallDestroyBehavior instance;

        private ToggleWallDestroyBehavior()
        {
        }

        public static ToggleWallDestroyBehavior Instance
        {
            get
            {
                lock (typeof(ToggleWallDestroyBehavior))
                {
                    return instance ?? (instance = new ToggleWallDestroyBehavior());
                }
            }
        }

        public virtual void Destroy(Block block)
        {
            Buttons.RemoveGreenButtonsListener(block);
        }
    }
}